hdat/hdat_cli.py

```diff
@@ -0,0 +1,113 @@
+import argparse
+import traceback
+
+from .resultspec import resolve_resultspecs, print_resultspec
+from .casespec import resolve_casespecs, select_suite
+from .runner import run_cases
+from .util import AbortError
+
+
+def parse_arguments(arguments):
+    parser = argparse.ArgumentParser(prog='hdat')
+    subparsers = parser.add_subparsers(dest='command', metavar='<command>')
+
+    run_help = 'run cases, store results in archive, compare against goldens'
+    run_parser = subparsers.add_parser('run', help=run_help)
+    run_parser.add_argument('casespecs', nargs='*', default=[''], metavar='<case>')
+    run_parser.add_argument('--collect-only', default=False, action='store_true')
+
+    show_help = 'visualize a single result'
+    show_parser = subparsers.add_parser('show', help=show_help)
+    show_result_help = 'result specifier to show'
+    show_parser.add_argument('resultspec', nargs="?", default='', metavar='<result>', help=show_result_help)
+
+    runshow_help = 'run then visualize a single result'
+    runshow_parser = subparsers.add_parser('runshow', help=runshow_help)
+    runshow_parser.add_argument('casespecs', nargs=1, default='', metavar='<result>')
+
+    diff_help = 'compare two results'
+    diff_parser = subparsers.add_parser('diff', help=diff_help)
+    diff_golden_help = 'result to compare to (defaults to current golden for the case)'
+    diff_parser.add_argument('goldenspec', nargs='?', default=None, metavar='<golden>', help=diff_golden_help)
+    diff_result_help = 'result being compared'
+    diff_parser.add_argument('resultspec', nargs=1, metavar='<result>', help=diff_result_help)
+
+    verify_help = 'move result metrics from archive to the golden store'
+    verify_parser = subparsers.add_parser('verify', help=verify_help)
+    verify_result_help = 'results to be stripped and moved into the golden store'
+    verify_parser.add_argument('resultspec', nargs='?', default='', metavar='<result>', help=verify_result_help)
+
+    return parser.parse_args(arguments)
+
+
+def _format_cases_status(cases_status):
+    return 'PASS: {}, FAIL: {}, UNKNOWN: {}, ERROR: {}'.format(
+        cases_status['pass'],
+        cases_status['fail'],
+        cases_status['unknown'],
+        cases_status['error'],
+    )
+
+
+def hdat_cli(arguments, suites, golden_store, archive, git_info):
+    args = parse_arguments(arguments)
+
+    if args.command is None:
+        parse_arguments(['-h'])
+
+    if args.command == 'run' and args.collect_only:
+        cases = resolve_casespecs(suites, args.casespecs)
+        print("\n".join(['{}/{}'.format(s, c) for s, c in cases]))
```

Comment:
I realize this was leftover from my code, but when I was reading this, I couldn't remember what `s` and `c` were, or the format that `resolve_casespecs` returns.  I think my code would have been better if I used:

```
print("\n".join(['{}/{}'.format(suite_id, case_id) for suite_id, case_id in cases]))
```

what do you think?

Reply:
Agreed